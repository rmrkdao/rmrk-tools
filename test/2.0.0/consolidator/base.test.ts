import { Consolidator } from "../../../src/rmrk2.0.0";
import {
  createBaseMock,
  createBaseMock2,
  getBlockCallsMock,
  getRemarksFromBlocksMock,
} from "../mocks";
import { cryptoWaitReady } from "@polkadot/util-crypto";

beforeAll(async () => {
  return await cryptoWaitReady();
});

describe("rmrk2.0.0 Consolidator: CREATE BASE", () => {
  it("should correctly create a Base", async () => {
    const remarks = getRemarksFromBlocksMock(
      getBlockCallsMock(createBaseMock().base())
    );
    const consolidator = new Consolidator();
    await consolidator.consolidate(remarks)
    const consolidatedResult = await consolidator.getResults()
    expect(consolidatedResult).toMatchSnapshot();
  });

  it("Should prevent creating a Base with duplicate part ids", async () => {
    const remarks = getRemarksFromBlocksMock(
      getBlockCallsMock(createBaseMock2().base())
    );
    const consolidator = new Consolidator();
    await consolidator.consolidate(remarks)
    const consolidatedResult = await consolidator.getResults()
    expect(consolidatedResult).toMatchSnapshot();
  });
});
