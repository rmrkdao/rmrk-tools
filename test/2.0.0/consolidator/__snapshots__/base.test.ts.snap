// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rmrk2.0.0 Consolidator: CREATE BASE Should prevent creating a Base with duplicate part ids 1`] = `
Object {
  "bases": Object {},
  "collections": Object {},
  "invalid": Array [
    Object {
      "block": 2,
      "caller": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "message": "[BASE] Duplicate base part id found: background",
      "object_id": "RMRK::BASE::2.0.0::{\\"symbol\\"%3A\\"KBASE777\\"%2C\\"type\\"%3A\\"svg\\"%2C\\"issuer\\"%3A\\"5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY\\"%2C\\"parts\\"%3A[{\\"id\\"%3A\\"background\\"%2C\\"type\\"%3A\\"slot\\"%2C\\"equippable\\"%3A[]%2C\\"z\\"%3A0}%2C{\\"id\\"%3A\\"background\\"%2C\\"type\\"%3A\\"slot\\"%2C\\"equippable\\"%3A[\\"d43593c715a56da27d-KANARIABIRDS\\"]%2C\\"z\\"%3A1}]%2C\\"themes\\"%3A{\\"themeOne\\"%3A{\\"_inherit\\"%3Atrue%2C\\"primary\\"%3A\\"%23fff\\"}}}",
      "op_type": "BASE",
    },
  ],
  "nfts": Object {},
}
`;

exports[`rmrk2.0.0 Consolidator: CREATE BASE should correctly create a Base 1`] = `
Object {
  "bases": Object {
    "base-2-KBASE777": Object {
      "block": 2,
      "changes": Array [],
      "id": "base-2-KBASE777",
      "issuer": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "metadata": undefined,
      "parts": Array [
        Object {
          "equippable": Array [],
          "id": "background",
          "type": "slot",
          "z": 0,
        },
        Object {
          "equippable": Array [
            "d43593c715a56da27d-KANARIABIRDS",
          ],
          "id": "backpack",
          "type": "slot",
          "z": 1,
        },
        Object {
          "id": "tail",
          "src": "ipfs://ipfs/QmcEuigDVCScMLs2dcrJ8qU4Q265xGisUyKeYdnnGFn6AE/var3_tail.svg",
          "type": "fixed",
          "z": 2,
        },
      ],
      "symbol": "KBASE777",
      "themes": Object {
        "themeOne": Object {
          "_inherit": true,
          "primary": "#fff",
        },
      },
      "type": "svg",
    },
  },
  "collections": Object {},
  "invalid": Array [],
  "nfts": Object {},
}
`;
